### PR TITLE
supplement(parser): reason for looping and searching for tags in the function isEnd is

### LINKED ISSUE #51: Chapter: top level await (script setup)

### LINKED ISSUE #213: supplement(parser): reason for looping and searching for tags in the function `isEnd` is
https://discord.com/channels/1190269119426215976/1190269542505644132/1191820837116981288

### PR DESCRIPTION
**Pull request by shoma-mano**
resolve #213 
I couldn't find clear reason for loop, so this tip is just [things ubugeeei said in discord](https://discord.com/channels/1190269119426215976/1190269627012493313/1191814911068754010) mostly.

### CODE REVIEW

File: book/online-book/src/10-minimum-example/070-more-complex-parser.md [hunk 1]
```diff
@@ -281,6 +281,27 @@ function startsWithEndTagOpen(source: string, tag: string): boolean {
...
+    for (let i = ancestors.length - 1; i >= 0; --i) {
+      if (startsWithEndTagOpen(s, ancestors[i].tag)) {
+        return true;
+      }
+    }
+  }
+
+  return !s;
+}
+```
```
Comment (Ubugeeei) [inline book/online-book/src/10-minimum-example/070-more-complex-parser.md:301]: Please remove semi-colons and double quotes.

**Review by Ubugeeei — CHANGES_REQUESTED** [review]
@shoma-mano 

Great explain!

Could you update `book/src/en` to be the same? 😄

File: book/online-book/src/10-minimum-example/070-more-complex-parser.md (hunk 1 above)
Comment (shoma-mano) [reply book/online-book/src/10-minimum-example/070-more-complex-parser.md]: I'm not sure my change that removes semi-colons and double quotes is ok, but I removed them anyway.

Comment (Ubugeeei) [reply book/online-book/src/10-minimum-example/070-more-complex-parser.md]: Yep! This is fine.
I made a comprehensive commit regarding this format issue yesterday, but it seems like it hasn't been reflected in your branch. (Please sync it.)

**Review by Ubugeeei — APPROVED** [review]

**Pull request merged**